packages/react-router/modules/Router.js

```diff
@@ -77,7 +78,10 @@ class Router extends React.Component {
 
   render() {
     const { children } = this.props;
-    return children ? React.Children.only(children) : null;
+    return RouterContext.provide(
+      this.getChildContext(),
```

Comment:
Maybe this should be `this.getChildContext().router` and then it can be consumed directly.
```js
const Cmp = props => (
  RouterContext.consume(router => <InnerCmp {...props} router={router} />)
);
```

Reply:
Tried that. Because of how the initial render works under this API, I ran into a boatload of errors.